### ISSUE TITLE
Sanitize plugin details from superglobals

### ISSUE THREAD

**Issue by GaryJones** [thomasgriffin/TGM-Plugin-Activation#345]
Starting around line 483, and 609 there were a couple of `@todo`'s:

~~~php
$plugin['name']   = $_GET['plugin_name']; // Plugin name. // @todo needs sanitizing, figure out how

$plugin['source'] = $_GET['plugin_source']; // Plugin source. // @todo needs sanitizing, figure out how
~~~

Line 1623:

~~~php
$plugins = (array) $_POST['plugin']; // @todo needs sanitizing, figure out how
~~~

And line 1804:

~~~
$plugins             = isset( $_POST['plugin'] ) ? (array) $_POST['plugin'] : array(); // @todo needs sanitizing, figure out how
~~~

May be handled via #287.

**Comment by jrfnl** [comment]
The simplest way might actually be to do a look-up in the `$this->plugins` array as what we are receiving should be registered in plugins array. Whitelist approach.

**Comment by jrfnl** [comment]
These should/will all be fixed by #368.

**Comment by jrfnl** [comment]
I'm pretty sure all of these have been addressed. Close ?

**Comment by GaryJones** [comment]
> Close ?

Only once I've merged #368 :-)

**Comment by jrfnl** [comment]
which was merged five days ago ?
